app/usecases/beatmap.py

```diff
@@ -261,8 +272,7 @@ def parse_from_osu_api(
             max_combo = 0
 
         ranked_status = RankedStatus.from_osu_api(int(response_json["approved"]))
-        if ranked_status in FROZEN_STATUSES:
-            frozen = True  # beatmaps are always frozen when ranked/approved/loved
```

Comment:
Looks like there was a bug here where frozen=True will get transferred to the next maps in the list, since it was set only one _before the loop_, rather than _for each iteration_. Doesn't necessarily line up with the behaviour I'd come to expect of the `save=` API, since individual maps might be ranked.